### PR TITLE
Added Support for StorageClasses

### LINKED ISSUE #34: Add support for CRD operation for StorageClasses.
Currently CITF does not support CRD operation for Storage classes. This support needs to be added. The code should also support reading/parsing SC definition from a yaml file .

Please refer to [this file](https://github.com/openebs/CITF/blob/master/utils/k8s/k8s.go) to see existing supported code.

**Please comment if you want to be assigned for this issue.**

Help links:
[Contributors guide](https://github.com/openebs/openebs/blob/master/CONTRIBUTING.md)
[hacktoberfest2018](https://openebs-community.slack.com/messages/hacktoberfest2018/) channel on slack. Ping us here if you want to discuss anything around hacktoberfest

### PR DESCRIPTION
**Pull request by mukeshmk**
Adds Get, List and Delete support for StorageClasses.
fixes #34 

Signed-off-by: Mukesh A <amukesh.mk@gmail.com>

### CODE REVIEW

File: utils/k8s/k8s.go [hunk 1]
```diff
@@ -660,3 +661,39 @@ func (k8s K8S) BlockUntilPodIsUpOrTimeout(pod *core_v1.Pod, timeout time.Duratio
...
+	logger.PrintlnDebugMessage(strings.Repeat("*", 80))
+	for _, storageClass := range storageClasses.Items {
+		logger.PrintlnDebugMessage("complete StorageClass name is:", storageClass.Name)
+		if strings.HasPrefix(storageClass.Name, storageClassPrefex) {
+			theStorageClasses = append(theStorageClasses, storageClass)
+		}
+	}
+	logger.PrintlnDebugMessage(strings.Repeat("*", 80))
+
+	return theStorageClasses, err
```
Comment (sonasingh46) [inline utils/k8s/k8s.go]: We can remove lines from 682.

**Comment by mukeshmk** [comment]
@sonasingh46 I have made changes as per the review comments.

File: utils/k8s/k8s.go [hunk 2]
```diff
@@ -660,3 +661,21 @@ func (k8s K8S) BlockUntilPodIsUpOrTimeout(pod *core_v1.Pod, timeout time.Duratio
...
+	return storageClassClient.Get(storageClassName, meta_v1.GetOptions{})
+}
+
+// ListStorageClasses returns all the StorageClass object.
+func (k8s K8S) ListStorageClasses() (*storage_v1.StorageClassList, error) {
+	storageClassClient := k8s.Clientset.StorageV1().StorageClasses()
+	return storageClassClient.List(meta_v1.ListOptions{})
+}
+
+// DeleteStorageClass deletes the StorageClass object for given storageClassName.
```
Comment (princerachit) [inline utils/k8s/k8s.go]: // DeleteStorageClass deletes the StorageClass object of given storageClassName.

File: utils/k8s/k8s.go [hunk 3]
```diff
@@ -660,3 +661,21 @@ func (k8s K8S) BlockUntilPodIsUpOrTimeout(pod *core_v1.Pod, timeout time.Duratio
...
 	return k8s.BlockUntilPodIsUpWithContext(ctx, pod)
 }
+
+// GetStorageClass returns the StorageClass object for given storageClassName.
+func (k8s K8S) GetStorageClass(storageClassName string) (*storage_v1.StorageClass, error) {
+	storageClassClient := k8s.Clientset.StorageV1().StorageClasses()
+	return storageClassClient.Get(storageClassName, meta_v1.GetOptions{})
+}
+
+// ListStorageClasses returns all the StorageClass object.
```
Comment (princerachit) [inline utils/k8s/k8s.go]: // ListStorageClasses returns a pointer to StorageClassList containing all the storage classes

**Pull request merged**